src/sentry/api/endpoints/organization_sdk_deprecations.py

```diff
@@ -0,0 +1,138 @@
+from sentry.models.projectsdk import EventType, ProjectSDK
+
+
+class SDKDeprecationsSerializer(serializers.Serializer):
+    event_type = serializers.ChoiceField(
+        choices=("profile",),
+        required=True,
+    )
+
+
+class SDKDeprecation(TypedDict):
+    projectId: str
+    minimumVersion: str
+    sdkName: str
+    sdkVersion: str
+
+
+class SDKDeprecationsResult(TypedDict):
+    data: list[SDKDeprecation]
+
+
+@region_silo_endpoint
+class OrganizationSdkDeprecationsEndpoint(OrganizationEndpoint):
+    owner = ApiOwner.PROFILING
+
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+
+    def get(self, request: Request, organization) -> Response:
+        serializer = SDKDeprecationsSerializer(data=request.GET)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        projects = self.get_projects(request, organization)
+
+        event_types = get_event_types(serializer.data["event_type"])
+
+        project_sdks = ProjectSDK.objects.filter(
+            project__in=projects,
+            event_type__in=[event_type.value for event_type in event_types],
+        )
+
+        sdk_deprecations_by_project: DefaultDict[Project, list[SDKDeprecation]] = defaultdict(list)
+        projects_with_up_to_date_sdk: set[Project] = set()
+
+        for project_sdk in project_sdks:
+            deprecation = get_deprecation_status(project_sdk)
+            if deprecation is None:
+                projects_with_up_to_date_sdk.add(project_sdk.project)
+            else:
+                sdk_deprecations_by_project[project_sdk.project].append(deprecation)
+
+        deprecations: list[SDKDeprecation] = []
+
+        for project, sdk_deprecations in sdk_deprecations_by_project.items():
+            if project in projects_with_up_to_date_sdk:
+                # It's possible that a single project contains data from different SDKs.
+                # Here we just want 1 up to date SDK sending data to this project.
+                #
+                # If we require all SDKs sending data to this project to be up to date,
+                # we risk checking against SDKs that may no longer be sending data.
+                # We would have to track a last seen timestamp to be more sure which
+                # is an expensive operation that we're not doing right now.
+                continue
+
+            deprecations.extend(sdk_deprecations)
+
+        result: SDKDeprecationsResult = {"data": deprecations}
+
+        return Response(result, status=200)
+
+
+def get_event_types(raw_event_type: str) -> list[EventType]:
+    if raw_event_type == "profile":
+        return [EventType.PROFILE_CHUNK]
+    raise ValueError(f"Unknown event type: {raw_event_type}")
+
+
+MINIMUM_SDK_VERSIONS: dict[tuple[int, str], str] = {
+    (EventType.PROFILE_CHUNK.value, "sentry.cocoa"): "8.49.0",
+    (EventType.PROFILE_CHUNK.value, "sentry.python"): "2.24.1",
+}
```

Comment:
should we make the values here options?

Reply:
Making them options would mean we'd need it to keep all the environments in sync which is very easily to overlook.

What are you trying to accomplish by making them options? Faster updates?

Reply:
@Zylphrex as long as the options are in the default `app.yaml`, they should be fine, no?

and yes, easier updates